File: core/src/main/java/co/ericp/freewayforecast/routes/Step.java
package co.ericp.freewayforecast.routes;

import java.util.List;

import co.ericp.freewayforecast.LatLng;

/**
 * One step of a trip (one stretch of driving without turns)
 */
public class Step {
    double distance;
    long duration;

    String instructions;

    LatLng start;
    LatLng end;

    List<LatLng> polyline;
}

File: core/src/main/java/co/ericp/freewayforecast/routes/Step.kt
package co.ericp.freewayforecast.routes

import co.ericp.freewayforecast.Location

/**
 * One step of a trip (one stretch of driving without turns)
 */
data class Step(
        val htmlInstructions: String,
        val distance: Long, // meters
        val duration: Long, // milliseconds
        val start: Location,
        val end: Location,
        val polyline: List<Location>
)
